projects/metis/src/app/_services/error.interceptor.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpErrorResponse,
+  HttpEvent,
+  HttpHandler,
+  HttpInterceptor,
+  HttpRequest
+} from '@angular/common/http';
+import { Router } from '@angular/router';
+import { Observable, retry, timer } from 'rxjs';
+import { RedirectPreviousUrl } from './redirect-previous-url.service';
+
+@Injectable({ providedIn: 'root' })
+export class ErrorInterceptor implements HttpInterceptor {
+  numberOfRetries = 2;
+  static retryDelay = 1000;
+
+  constructor(
+    private readonly router: Router,
+    private readonly redirectPreviousUrl: RedirectPreviousUrl
+  ) {}
+
+  /** intercept
+   * Adds a retry pipe to the http handler
+   * @param { HttpRequest } request
+   * @param { HttpHandler } next
+   * @returns Observable<HttpEvent>
+   **/
+  intercept(request: HttpRequest<unknown>, next: HttpHandler): Observable<HttpEvent<unknown>> {
+    return next
+      .handle(request)
+      .pipe(retry({ count: this.numberOfRetries, delay: this.shouldRetry }));
+  }
+
+  /** shouldRetry
+   * Returns a predfined delay or throws the error passed
+   * @param { HttpErrorResponse } error
+   * @returns Observable<number>
+   **/
+  shouldRetry(error: HttpErrorResponse): Observable<number> {
+    const status = parseInt(`${error.status}`);
+    if (status === 401) {
+      this.expiredToken();
+    } else if (![200, 406].includes(status)) {
```

Comment:
Why are we specifying the http status we are looking for?
Don't we want to generalise and check for 2xx and 4xx statuses?

Reply:
200 (success) is not the same as 204 (no content).  We care about what should behave differently, and I've kept the relevant codes the same as they were.